### PR TITLE
Suppressing warnings generated by 5 GraphBolt tests.

### PR DESCRIPTION
**Pull request by drivanov**
## Description

Elimination of the following warnings:
```
tests/python/pytorch/graphbolt/test_base.py: 2 warnings
tests/python/pytorch/graphbolt/test_feature_fetcher.py: 4 warnings
tests/python/pytorch/graphbolt/test_item_sampler.py: 9 warnings
tests/python/pytorch/graphbolt/test_base.py::test_CopyTo 1 warning
  /usr/local/lib/python3.10/dist-packages/dgl/graphbolt/item_sampler.py:45: DGLWarning: Failed to map item list to `MiniBatch` as the names of items are not provided. Please provide a customized `MiniBatcher`. The item list is returned as is.
    dgl_warning(
```

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: tests/python/pytorch/graphbolt/test_base.py [hunk 1]
```diff
@@ -13,17 +14,20 @@
...
+    def checkData(dp):
+        # Filtering DGLWarning:
+        #    Failed to map item list to `MiniBatch`
+        #    as the names of items are not provided.
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", category=UserWarning)
+            for data in dp:
+                assert data.device.type == "cuda"
 
+    item_sampler = gb.ItemSampler(gb.ItemSet(torch.randn(20)), 4)
```
Comment (Rhett-Ying) [inline tests/python/pytorch/graphbolt/test_base.py:26]: I suggest to add `names` for `ItemSet` to eliminate warnings from the root instead of filtering.

Comment (drivanov) [reply tests/python/pytorch/graphbolt/test_base.py:26]: I am sorry but with that changes
```
 item_sampler = gb.ItemSampler(gb.ItemSet(torch.randn(20), names="seed_nodes"), 4)
```
the `test_CopyTo` fails:
```
=============================================================================================== FAILURES ================================================================================================
______________________________________________________________________________________________ test_CopyTo ______________________________________________________________________________________________

    @unittest.skipIf(F._default_context_str == "cpu", "CopyTo needs GPU to test")
    def test_CopyTo():
        item_sampler = gb.ItemSampler(gb.ItemSet(torch.randn(20), names="seed_nodes"), 4)
        #item_sampler = gb.ItemSampler(gb.ItemSet(torch.randn(20)), 4)
        import pdb; pdb.set_trace()
   
        # Invoke CopyTo via class constructor.
        dp = gb.CopyTo(item_sampler, "cuda")
        for data in dp:
>           assert data.device.type == "cuda"
E           AttributeError: 'MiniBatch' object has no attribute 'device'

tests/python/pytorch/graphbolt/test_base.py:21: AttributeError
```
This is what it looks like in `pdb`:
```
> /opt/dgl/dgl-source/tests/python/pytorch/graphbolt/test_base.py(21)test_CopyTo()
-> assert data.device.type == "cuda"
(Pdb) dir(data.device)
*** AttributeError: 'MiniBatch' object has no attribute 'device'
(Pdb)
```
Without adding `names` to  `ItemSet`
```
item_sampler = gb.ItemSampler(gb.ItemSet(torch.randn(20)), 4)
```
this test has no problems, and in the same line we can see::
```
> /opt/dgl/dgl-source/tests/python/pytorch/graphbolt/test_base.py(21)test_CopyTo()
-> assert data.device.type == "cuda"
(Pdb) dir(data.device)
['__class__', '__delattr__', '__dir__', '__doc__', '__enter__', '__eq__', '__exit__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', 'index', 'type']
(Pdb)
```

**Comment by frozenbugs** [comment]
@dgl-bot

File: tests/python/pytorch/graphbolt/test_base.py [hunk 2]
```diff
@@ -13,17 +14,20 @@
 
 @unittest.skipIf(F._default_context_str == "cpu", "CopyTo needs GPU to test")
 def test_CopyTo():
-    item_sampler = gb.ItemSampler(gb.ItemSet(torch.randn(20)), 4)
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/test_base.py:16]: Actually the original code is not a suggested way to use ItemSampler, to address the warning related to minibatch.

File: tests/python/pytorch/graphbolt/test_base.py [hunk 3]
```diff
@@ -13,17 +14,20 @@
...
     # Invoke CopyTo via class constructor.
-    dp = gb.CopyTo(item_sampler, "cuda")
-    for data in dp:
-        assert data.device.type == "cuda"
-
+    checkData(gb.CopyTo(item_sampler, "cuda"))
     # Invoke CopyTo via functional form.
-    dp = item_sampler.copy_to("cuda")
-    for data in dp:
-        assert data.device.type == "cuda"
```
[the remark this answers is not in the document]
Comment (drivanov) [reply tests/python/pytorch/graphbolt/test_base.py]: Done. Thank you!

**Comment by Rhett-Ying** [comment]
@drivanov I made a fix to eliminate the warnings in https://github.com/dmlc/dgl/pull/7054

**Pull request closed without merging**